### ISSUE TITLE
strip-indent causing Maximum call stack size exceeded

### ISSUE THREAD

**Issue by nathancahill** [sveltejs/svelte-preprocess#143]
**Describe the bug**
I'm using `svelte-preprocess` with PostCSS and Tailwind CSS, so the generated styles in local development is huge (3.2mb). The `strip-indent` library is throwing `RangeError: Maximum call stack size exceeded` with the full style sheet. If I cut it down to around 1.5mb the error goes away.

If I remove the two calls to `strip-indent` in `svelte-preprocess`, the error also goes away.

**To Reproduce**
Add PostCSS and Tailwind CSS to a Svelte project and compile:

```html
<style global>
  @tailwind base;
  @tailwind components;
  @tailwind utilities;
</style>
```

**Expected behavior**
`svelte-preprocess` should be able to handle 3.2mb of CSS.

**Stacktraces**

<details>
  <summary>Stack trace</summary>

```js
[!] (plugin svelte) RangeError: Maximum call stack size exceeded
src/App.svelte
RangeError: Maximum call stack size exceeded
    at module.exports (~/node_modules/min-indent/index.js:10:18)
    at Object.module.exports [as default] (~/node_modules/strip-indent/index.js:5:17)
    at Object.exports.runTransformer (~/node_modules/svelte-preprocess/dist/utils.js:1:2279)
    at style (~/node_modules/svelte-preprocess/dist/autoProcess.js:1:2824)
    at ~/node_modules/svelte/src/compiler/preprocess/index.ts:122:34
    at async Promise.all (index 0)
    at replace_async (~/node_modules/svelte/src/compiler/preprocess/index.ts:60:48)
    at preprocess (~/node_modules/svelte/src/compiler/preprocess/index.ts:115:12)
    at async Promise.all (index 0)
    at async Promise.all (index 0)
```

</details>

**Information about your project:**

- Operating system: OS X
- `svelte-preprocess` version: v3.7.4
- Bundled with Rollup

**Comment by benmccann** [comment]
@nathancahill could you also file an issue over at strip-indent? It'd also be helpful if you could provide them the 3.2mb CSS file

**Comment by nathancahill** [comment]
Finally had some time to dive in to this issue.

[`strip-indent`](https://github.com/sindresorhus/strip-indent/blob/master/index.js#L2) depends on [`min-indent`](https://github.com/jamiebuilds/min-indent/blob/master/index.js) to determine the shortest leading whitespace from lines in a string.
When running `min-indent` on a 3.2mb CSS file, the regex [`/^[ \t]*(?=\S)/gm`](https://github.com/jamiebuilds/min-indent/blob/master/index.js#L3) produces 150376 matches.
The problem line is the [last one](https://github.com/jamiebuilds/min-indent/blob/master/index.js#L10):

```js
return Math.min.apply(Math, match.map(x => x.length));
```

This maps over every indentation match to get it's length, then calls `Math.min` on
the resulting array. When called with 150376 matches it throws `RangeError: Maximum call stack size exceeded`.

Instead of using the `Math` construction, a simpler method uses `reduce` with no call stack error:

```js
return match.reduce((r, a) => Math.min(r, a.length), Infinity)
```

Linking to this issue on `min-indent`.

**Comment by nathancahill** [comment]
Resolved in `min-indent@1.0.1`
